pkg/logs/launchers/container/tailerfactory/file_test.go

```diff
@@ -117,7 +117,7 @@ func TestMakeFileSource_docker_success(t *testing.T) {
 		AutoMultiLineSampleSize:     123,
 		AutoMultiLineMatchThreshold: 0.123,
 	})
-	child, err := tf.makeFileSource(source)
```

Comment:
Can we add a couple of unit tests to cover the new types of config conversions? You can either rename this file or create a new `socket_test.go`. The existing tests should be a good guide